x-pack/solutions/search/plugins/enterprise_search/public/applications/applications/components/search_application/search_application_view_index_link.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import { EuiLink } from '@elastic/eui';
+
+import { KibanaLogic } from '../../../shared/kibana';
+
+export const SearchApplicationViewIndexLink: React.FC<{
+  indexName: string;
+  dataTelemetryId?: string;
+  target?: boolean;
+}> = ({ indexName, dataTelemetryId }) => {
+  const { share } = useValues(KibanaLogic);
+
+  const searchIndexDetailsUrl: string =
+    share?.url.locators.get('SEARCH_INDEX_DETAILS_LOCATOR_ID')?.useUrl({ indexName }) ?? '';
+
+  return searchIndexDetailsUrl ? (
+    <EuiLink
+      data-telemetry-id={dataTelemetryId}
+      data-test-subj={'search-application-index-link'}
```

Comment:
Should this be a prop along with `data-telemetry-id` ?

Reply:
We could  added here  as `data-test-subj` value is same in [both usage](https://github.com/search?q=repo%3Aelastic%2Fkibana+search-application-index-link&type=code) (flyout & search application content).

Reply:
updated 359b77ee2b258d5a6b7751e9cb6748f490be0200